sdk/src/main/java/com/mapbox/maps/StyleObserver.kt

```diff
@@ -13,51 +13,60 @@ import java.util.concurrent.CopyOnWriteArrayList
  * and maintains and invokes user added listeners.
  */
 internal class StyleObserver(
-  private val mapboxMap: MapboxMap,
+  private val nativeMapWeakRef: WeakReference<MapInterface>,
+  private val styleLoadedListener: Style.OnStyleLoaded,
   private val nativeObserver: NativeObserver,
   private val pixelRatio: Float
 ) : OnStyleLoadedListener, OnMapLoadErrorListener {
 
-  private val awaitingStyleLoadListeners = CopyOnWriteArrayList<Style.OnStyleLoaded>()
-  private var awaitingStyleErrorListener: OnMapLoadErrorListener? = null
+  @Volatile
```

Comment:
our maps sdk is single-threaded, why volatile here and below?

Reply:
Hm, is onStyleLoaded called from the same thread? 
Why was `CopyOnWriteArrayList` used previously?

Reply:
IDK
however even if callbacks arrive from different threads (although I doubt it) - running that code from this collection will most likely lead to a problem

Reply:
Removed, callback arrives on the main thread